zircon.core/common/src/main/kotlin/org/hexworks/zircon/internal/component/impl/DefaultNumberInput.kt

```diff
@@ -0,0 +1,259 @@
+package org.hexworks.zircon.internal.component.impl
+
+import org.hexworks.cobalt.databinding.api.createPropertyFrom
+import org.hexworks.cobalt.events.api.Subscription
+import org.hexworks.cobalt.logging.api.LoggerFactory
+import org.hexworks.zircon.api.behavior.ChangeListener
+import org.hexworks.zircon.api.behavior.Disablable
+import org.hexworks.zircon.api.behavior.Scrollable
+import org.hexworks.zircon.api.builder.component.ComponentStyleSetBuilder
+import org.hexworks.zircon.api.builder.graphics.StyleSetBuilder
+import org.hexworks.zircon.api.color.TileColor
+import org.hexworks.zircon.api.component.ColorTheme
+import org.hexworks.zircon.api.component.ComponentStyleSet
+import org.hexworks.zircon.api.component.NumberInput
+import org.hexworks.zircon.api.component.TextArea
+import org.hexworks.zircon.api.component.data.ComponentMetadata
+import org.hexworks.zircon.api.component.renderer.ComponentRenderingStrategy
+import org.hexworks.zircon.api.data.Position
+import org.hexworks.zircon.api.extensions.whenEnabled
+import org.hexworks.zircon.api.extensions.whenEnabledRespondWith
+import org.hexworks.zircon.api.uievent.KeyCode
+import org.hexworks.zircon.api.uievent.KeyboardEvent
+import org.hexworks.zircon.api.uievent.KeyboardEventType
+import org.hexworks.zircon.api.uievent.MouseEvent
+import org.hexworks.zircon.api.uievent.Pass
+import org.hexworks.zircon.api.uievent.Processed
+import org.hexworks.zircon.api.uievent.UIEventPhase
+import org.hexworks.zircon.api.util.TextUtils
+import org.hexworks.zircon.internal.Zircon
+import org.hexworks.zircon.internal.behavior.impl.DefaultScrollable
+import org.hexworks.zircon.internal.component.impl.textedit.EditableTextBuffer
+import org.hexworks.zircon.internal.component.impl.textedit.cursor.MovementDirection.LEFT
+import org.hexworks.zircon.internal.component.impl.textedit.cursor.MovementDirection.RIGHT
+import org.hexworks.zircon.internal.component.impl.textedit.transformation.DeleteCharacter
+import org.hexworks.zircon.internal.component.impl.textedit.transformation.DeleteCharacter.DeleteKind.BACKSPACE
+import org.hexworks.zircon.internal.component.impl.textedit.transformation.DeleteCharacter.DeleteKind.DEL
+import org.hexworks.zircon.internal.component.impl.textedit.transformation.InsertCharacter
+import org.hexworks.zircon.internal.component.impl.textedit.transformation.MoveCursor
+import org.hexworks.zircon.internal.event.ZirconEvent
+import org.hexworks.zircon.internal.event.ZirconScope
+import kotlin.math.min
+
+abstract class DefaultNumberInput(
```

Comment:
This is an `abstract` class so we can't really use it as a default implementation to delegate to (like with the other components). We usually name classes like these base classes, somewhere along the lines of `BaseNumberInput`.